src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -144,6 +153,13 @@ public static void ApplyCentralVersionInformation(IList<LibraryDependency> packa
             {
                 foreach (LibraryDependency d in packageReferences.Where(d => !d.AutoReferenced && d.LibraryRange.VersionRange == null))
                 {
+                    if (d.VersionOverride != null)
+                    {
+                        d.LibraryRange.VersionRange = d.VersionOverride;
```

Comment:
Do we ever use the overriden information? 

Would we be better off just adding a property that tells us we have an override instead?

Reply:
It basically seems less error to do it from the onset instead of replacing it here.

Reply:
We need to be able to distinguish between when a user specified a `Version` (to emit an error and just `VersionOverride`.  At the moment, that information is gathered from a source (project system, MSBuild) and we create a `ProjectSpec`.  When processing the `ProjectSpec` as part of restore is when this information is processed.

To do it the other way around, I'd need to update the code in the following data gathering code paths which construct the `ProjectSpec`:
* Legacy project system
* CPS
* MSBuild/dotnet GenerateRestoreGraph
* MSBuild/dotnet Static graph

It felt simpler at the time to leave those systems as just gathering the data for processing later.  If you feel strongly, I can update the code to apply the versions when gathering the data and emit any errors/warnings there.

Reply:
Oh I understand why we need the information that the version has been overriden. 

My suggestion is instead of Version and VersionOverride to have, Version, where it's the effective version and OriginalVersion, or a bool to indicate the the version has been changed, such as VersionOverriden. 

That way you don't need to rewrite the versionn range and you're less likely to get into funny business because the majority of the restore code still just uses, Version.

Reply:
Resolving this.

Reply:
> My suggestion is instead of Version and VersionOverride to have, Version, where it's the effective version and OriginalVersion, or a bool to indicate the the version has been changed, such as VersionOverriden.

I am thinking the same. Given the comment has been resolved, I am curious to learn more incase if I missed any context.